lib/models/hostapp.ts

```diff
@@ -0,0 +1,256 @@
+				osVersions = osVersions.concat(
+					getOsVersionsFromReleases(hostApp.owns__release ?? [], appTags),
+				);
+				deviceTypeOsVersions[hostAppDeviceType] = osVersions;
+
+				return deviceTypeOsVersions;
+			},
+			{},
+		);
+
+		return res;
+	};
+
+	// This mutates the passed object.
+	const transformVersionSets = (deviceTypeOsVersions: DeviceTypeOsVersions) => {
+		Object.keys(deviceTypeOsVersions).forEach((deviceType) => {
+			deviceTypeOsVersions[deviceType].sort(sortVersions);
+			const recommendedPerOsType: Dictionary<boolean> = {};
+
+			// Note: the recommended version settings might come from the server in the future, for now we just set it to the latest version for each os type.
+			deviceTypeOsVersions[deviceType].forEach((version) => {
+				if (!recommendedPerOsType[version.osType]) {
+					if (
+						version.variant !== 'dev' &&
+						!bSemver.prerelease(version.rawVersion)
+					) {
+						const additionalFormat = version.line
+							? ` (${version.line}, recommended)`
+							: ' (recommended)';
+
+						version.isRecommended = true;
+						version.formattedVersion = `v${version.strippedVersion}${additionalFormat}`;
+						recommendedPerOsType[version.osType] = true;
+					}
+				}
+			});
+		});
+
+		return deviceTypeOsVersions;
+	};
+
+	const getOsVersions = (deviceTypes: string[]) => {
+		return pine.get<AppWithDeviceType>({
+			resource: 'application',
+			options: {
+				$filter: {
+					is_host: true,
+					is_for__device_type: {
+						$any: {
+							$alias: 'dt',
+							$expr: {
+								dt: {
+									slug: { $in: deviceTypes },
+								},
+							},
+						},
+					},
+				},
+				$select: ['id', 'app_name'],
+				$expand: {
+					application_tag: {
+						$select: ['id', 'tag_key', 'value'],
+					},
+					is_for__device_type: {
+						$select: ['slug'],
+					},
+					owns__release: {
+						$select: ['id'],
+						$expand: {
+							release_tag: {
+								$select: ['id', 'tag_key', 'value'],
+							},
+						},
+						$filter: {
+							is_invalidated: false,
+						},
+					},
+				},
+			},
+		});
+	};
+
+	const memoizedGetOsVersions = memoizee(getOsVersions, {
+		maxAge: 10 * 60 * 1000,
+		primitive: true,
+		promise: true,
+	});
+
+	/**
+	 * @summary Get all OS versions for the passed device types
+	 * @name getAllOsVersions
+	 * @public
+	 * @function
+	 * @memberof balena.models.hostapp
+	 *
+	 * @param {String[]} deviceTypes - device type slugs
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.hostapp.getALlOsVersions(['fincm3', 'raspberrypi3']);
+	 */
+	const getAllOsVersions = async (
+		deviceTypes: string[],
+	): Promise<DeviceTypeOsVersions> => {
+		const sortedDeviceTypes = deviceTypes.sort();
+		const hostapps = await memoizedGetOsVersions(sortedDeviceTypes);
+		const deviceTypeOsVersions = transformVersionSets(
+			transformHostApps(hostapps),
+		);
+		Object.values(deviceTypeOsVersions).forEach((versions) => {
+			versions.sort(sortVersions);
+		});
```

Comment:
Is this already done in `transformVersionSets` by `deviceTypeOsVersions[deviceType].sort(sortVersions);` ?

Reply:
Yeah, it seems it was a mistake in the dashboard, I just copied it over 👍